### PR TITLE
documentation(834649): Emoji picker module in rich text editor Documentation

### PR DESCRIPTION
**Pull request by venkateshsf3956**
Emoji picker module in rich text editor Documentation

### CODE REVIEW

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        }]
+    }
+});
+defaultRTE.appendTo('#defaultRTE');
+
+```
+
+In addition to using the `iconCSS` property to define the style for the emoji icon in the heading of the icon, you also have the option to provide custom code for the icon itself.
+
+If you prefer to use a specific icon, you can include the corresponding HTML or CSS code within the emoji picker heading. This allows you to display a custom icon or choose from a library of icons to represent the emoji picker.
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:95]: If you prefer to use a specific icon, you can include the corresponding HTML or CSS code within the emoji picker heading. This allows you to display a custom icon or choose from a library of icons to represent the emoji picker.


reframe this content. not clear it

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+By combining the flexibility of the `iconCSS` property with the ability to provide custom code for the icon, you can achieve a highly customized and visually appealing emoji picker heading in the Rich Text Editor.
+
+> Rich Text Editor features are segregated into individual feature-wise modules. To use emoji picker, inject the Emoji picker module using the `RichTextEditor.Inject(Audio)`.
+
+
+## Using the shortcut key to open the emoji picker
+
+In the Rich Text Editor, when you type a word prefix or start typing a word in the editor, followed by a colon (":"), the emoji picker popup will open. This functionality allows you to easily access the emoji picker by simply typing a word and using a colon.
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:104]: You can enable the instant display of the emoji picker by pressing the colon (:) key when typing a word prefix in an editor.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+platform: ej2-javascript
+control: Emoji picker
+publishingplatform: ##Platform_Name##
+documentation: ug
+domainurl: ##DomainURL##
+---
+
+# Emoji picker in ##Platform_Name## Rich text editor control
+
+A Rich Text Editor is a tool that allows users to easily add emojis or emoticons to their text. Typically, it is a small window or panel that displays a variety of emojis, arranged in different categories, such as smileys, animals, food, and so on. Users can select the desired emoji by clicking on it or by typing its name in a search bar.
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:14]: An emoji picker is a tool

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+We have provided another possibility to open the emoji picker popup by using the `showEmojiPicker()` method.
+
+If you call the `showEmojiPicker()` method, it automatically adjusts its position based on the cursor's location.
+
+Alternatively, you can modify the position of the popup by calling the `showEmojiPicker()` method and specifying the `x` and `y` parameters.
+
+`x` - Left position.
+`y` - Top position.
+
+This allows you to have precise control over where the emoji picker appears on the screen.
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:142]: You can display the emoji picker anywhere within the editor by using the showEmojiPicker public method. By default, it opens at the current cursor position within the editor. If you wish to customize the position of the emoji picker, you can specify the x and y parameters when calling this method.


Sample?

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+# Emoji picker in ##Platform_Name## Rich text editor control
+
+A Rich Text Editor is a tool that allows users to easily add emojis or emoticons to their text. Typically, it is a small window or panel that displays a variety of emojis, arranged in different categories, such as smileys, animals, food, and so on. Users can select the desired emoji by clicking on it or by typing its name in a search bar.
+
+## Enabling the toolbar option and custom emojis.
+
+You can add the `EmojiPicker` tool to the toolbar of the Rich Text Editor by utilizing the `toolbarSettings` [items](../api/rich-text-editor/toolbarSettings/#items) property.
+
+To customize the emojis in the Rich Text Editor, you can utilize the following list of options available in the [`emojiPickerSettings`]
+
```
[the remark this answers is not in the document]
Comment (thangavele) [reply ej2-javascript/rich-text-editor/ts/emoji-picker.md:21]: By default, a predefined set of emojis is configured. However, you have the flexibility to customize these icons according to your needs. To achieve this, you can utilize the emojiPickerSettings property.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        { code: '1F6AD', desc: 'No smoking' }]
+        }]
+    }
+});
+defaultRTE.appendTo('#defaultRTE');
+
+```
+
+In addition to using the [iconCss](../api/rich-text-editor/emojiPickerSettings/iconCss) property as the first priority for defining the style of the emoji icon in the heading, you also have the option to provide custom code for the icon itself. If the `iconCss` property is not needed, you can utilize the [code](../api/rich-text-editor/emojiPickerSettings/code) property which represents the Unicode of the icon displayed in the emoji picker toolbar item.
+
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:97]: Additionally, you have the option to customize the icons of toolbar items using the iconCSS and code properties. The iconCSS property allows you to define a custom CSS class for the toolbar item icon, while the code property enables you to specify the Unicode character code for the icon.

When both iconCSS and code properties are provided, the iconCSS property takes precedence in determining the appearance of the toolbar item icon.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+{% endtabs %}
+          
+{% previewsample "page.domainurl/code-snippet/rich-text-editor/emoji-picker-cs1" %}
+
+> RichTextEditor features are segregated into individual feature-wise modules. To use emoji picker, inject the Emoji Picker module using the `RichTextEditor.Inject(EmojiPicker)`.
+
+## Using the shortcut key to open the emoji picker
+
+You can enable the instant display of the emoji picker by pressing the colon (:) key when typing a word prefix in an editor. You don't need to worry about filtering the particular emoji without using a text box. Simply filter the emoji by typing a colon after the name of the emoji, and it will filter the emojis accordingly.
+
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:118]: You can quickly access the emoji picker by simply pressing the colon (:) key while typing a word prefix in an editor, allowing for instant emoji selection and display. Moreover, you can continue typing in the editor after the colon (:) to filter and refine your search for the desired emojis.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+## Opening the emoji picker in a custom position
+
+You can display the emoji picker anywhere within the editor by using the `showEmojiPicker` public method. By default, it opens at the current cursor position within the editor. If you wish to customize the position of the popup by calling the `showEmojiPicker()` method and specifying the `x` and `y` parameters.
+
+`x` - Left position.
+`y` - Top position.
+
+You can customize the horizontal and vertical position by specifying a value as 'center', 'auto', or a pixel-based value to achieve the desired placement. This allows you to have precise control over where the emoji picker appears on the screen.
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:139]: horizontal and vertical 
 
 top and left position

Comment (thangavele) [reply ej2-javascript/rich-text-editor/ts/emoji-picker.md:139]: In the following sample, customize the left and top positions by specifying a value as 'center' and 'auto' in the showEmojiPicker method when pressing the slash (/) key within an editor.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+When you type '/' in the editor, the popup will open at the specified position as shown in the code examples below.
+
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:144]: use the content alone

File: ej2-javascript/code-snippet/rich-text-editor/emoji-picker-cs2/index.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+        'UnorderedList', '|', 'CreateLink', 'Image', '|', 'SourceCode', 'EmojiPicker',
+        '|', 'Undo', 'Redo'] 
+    },
+});
+
+emojiPickerRTE.appendTo('#emojipickerRTE');
+
+emojiPickerRTE.addEventListener('keydown', (event: KeyboardEvent) => {
+    if (event.keyCode === 191) {
+        emojiPickerRTE.showEmojiPicker('center',100);
```
Comment (thangavele) [inline ej2-javascript/code-snippet/rich-text-editor/emoji-picker-cs2/index.ts:16]: center and auto

File: ej2-javascript/rich-text-editor/js/emoji-picker.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+    defaultRTE.appendTo('#defaultRTE');
+</script>
+
+```
+
+Additionally, you have the option to customize the icons of toolbar items using the [iconCss](../api/rich-text-editor/emojiPickerSettings/iconCss) and [code](../api/rich-text-editor/emojiPickerSettings/code) properties. The `iconCSS` property allows you to define a custom CSS class for the toolbar item icon, while the `code` property enables you to specify the Unicode character code for the icon.
+
+When both `iconCSS` and `code` properties are provided, the `iconCSS` property takes precedence in determining the appearance of the toolbar item icon.
+
+Additionally, you have the option to enhance the user experience by implementing a filtering feature for efficiently managing a large dataset of emojis. By setting the [showSearchBox](../api/rich-text-editor/emojiPickerSettings/showSearchBox) parameter to true (which is the default value), users will be able to utilize a search box to filter the displayed emojis according to their preferences.
```
Comment (thangavele) [inline ej2-javascript/rich-text-editor/js/emoji-picker.md:103]: parameter to property

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md (hunk 9 above)
Comment (thangavele) [reply ej2-javascript/rich-text-editor/ts/emoji-picker.md:144]: In the following sample, customize the left and top positions by specifying a value as 'center' and 'auto' in the showEmojiPicker method when pressing the slash (/) key within an editor.

File: ej2-javascript/rich-text-editor/js/emoji-picker.md [hunk 12]
```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: Emoji Picker in ##Platform_Name## RichTextEditor | Syncfusion
+description: Learn here all about Emoji Picker in Syncfusion ##Platform_Name## RichTextEditor control of Syncfusion Essential JS 2 and more.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/js/emoji-picker.md:4]: description: Learn about Emoji Picker in Syncfusion ##Platform_Name## RichTextEditor control of Syncfusion Essential JS 2 and more.

File: ej2-javascript/rich-text-editor/js/emoji-picker.md [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+platform: ej2-javascript
+control: Emoji Picker
+publishingplatform: ##Platform_Name##
+documentation: ug
+domainurl: ##DomainURL##
+---
+
+# Emoji Picker in ##Platform_Name## RichTextEditor control
+
+An emoji picker is a tool that allows users to easily add emojis or emoticons to their text. Typically, it is a small window or panel that displays a variety of emojis, arranged in different categories, such as smileys, animals, food, and so on. Users can select the desired emoji by clicking on it or by typing its name in a search bar.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/js/emoji-picker.md:14]: An emoji picker is a tool that allows users to add emojis or emoticons to their text easily. Typically, it is a small window or panel that displays a variety of emojis arranged in different categories, such as smileys, animals, food, and so on. Users can select the desired emoji by clicking on it or typing its name in a search bar.

File: ej2-javascript/rich-text-editor/js/emoji-picker.md [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+
+## Enabling the toolbar option and custom emojis.
+
+You can add the `EmojiPicker` tool to the toolbar of the RichTextEditor by utilizing the `toolbarSettings` [items](../api/rich-text-editor/toolbarSettings/#items) property.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/js/emoji-picker.md:18]: Add the `EmojiPicker` tool to the toolbar of the RichTextEditor by utilizing the `toolbarSettings` [items](../api/rich-text-editor/toolbarSettings/#items) property.

File: ej2-javascript/rich-text-editor/js/emoji-picker.md [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+
+By default, a predefined set of emojis is configured. However, you have the flexibility to customize these icons according to your needs. To achieve this, you can utilize the [emojiPickerSettings](../api/rich-text-editor/emojiPickerSettings/) property.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/js/emoji-picker.md:20]: By default, a predefined set of emojis is configured. However, you can customize these icons according to your needs. To achieve this, utilize the [emojiPickerSettings](../api/rich-text-editor/emojiPickerSettings/) property.

File: ej2-javascript/rich-text-editor/js/emoji-picker.md [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+{% highlight html tabtitle="index.html" %}
+{% include code-snippet/rich-text-editor/emoji-picker-cs1/index.html %}
+{% endhighlight %}
+{% endtabs %}
+          
+{% previewsample "page.domainurl/code-snippet/rich-text-editor/emoji-picker-cs1" %}
+
+## Using the shortcut key to open the emoji picker
+
+You can quickly access the emoji picker by simply pressing the colon (:) key while typing a word prefix in an editor, allowing for instant emoji selection and display. Moreover, you can continue typing in the editor after the colon (:) to filter and refine your search for the desired emojis.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/js/emoji-picker.md:120]: Quickly access the emoji picker by simply pressing the colon (:) key while typing a word prefix in an editor, allowing for instant emoji selection and display. Moreover, continue typing in the editor after the colon (:) to filter and refine your search for the desired emojis.

File: ej2-javascript/rich-text-editor/js/emoji-picker.md [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+`Arrow keys`: Use the arrow keys (up, down, left, right) to move the emoji focus in the corresponding direction.
+
+`Enter`: Press Enter key to select the currently focused emoji.
+
+`Escape`: Press Escape to close the emoji picker popup without selecting an emoji.
+
+
+## Opening the emoji picker in a custom position
+
+You can display the emoji picker anywhere within the editor by using the `showEmojiPicker` public method. By default, it opens at the current cursor position within the editor. If you wish to customize the position of the popup by calling the `showEmojiPicker()` method and specifying the `x` and `y` parameters.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/js/emoji-picker.md:137]: Display the emoji picker anywhere within the editor by using the `showEmojiPicker` public method. By default, it opens at the current cursor position within the editor. If you wish to customize the position of the popup, call the `showEmojiPicker()` method and specify the `x` and `y` parameters.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 18]
```diff
@@ -0,0 +1,152 @@
+---
+layout: post
+title: Emoji Picker in ##Platform_Name## RichTextEditor | Syncfusion
+description: Learn here all about Emoji Picker in Syncfusion ##Platform_Name## RichTextEditor control of Syncfusion Essential JS 2 and more.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:4]: Description: Learn about Emoji Picker in Syncfusion ##Platform_Name## RichTextEditor control of Syncfusion Essential JS 2 and more.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+platform: ej2-javascript
+control: Emoji Picker
+publishingplatform: ##Platform_Name##
+documentation: ug
+domainurl: ##DomainURL##
+---
+
+# Emoji Picker in ##Platform_Name## RichTextEditor control
+
+An emoji picker is a tool that allows users to easily add emojis or emoticons to their text. Typically, it is a small window or panel that displays a variety of emojis, arranged in different categories, such as smileys, animals, food, and so on. Users can select the desired emoji by clicking on it or by typing its name in a search bar.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:14]: An emoji picker is a tool that allows users to add emojis or emoticons to their text easily. Typically, it is a small window or panel that displays a variety of emojis arranged in different categories, such as smileys, animals, food, and so on. Users can select the desired emoji by clicking on it or by typing its name in a search bar.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+
+## Enabling the toolbar option and custom emojis.
+
+You can add the `EmojiPicker` tool to the toolbar of the RichTextEditor by utilizing the `toolbarSettings` [items](../api/rich-text-editor/toolbarSettings/#items) property.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:18]: Add the `EmojiPicker` tool to the toolbar of the RichTextEditor by utilizing the `toolbarSettings` [items](../api/rich-text-editor/toolbarSettings/#items) property.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+
+By default, a predefined set of emojis is configured. However, you have the flexibility to customize these icons according to your needs. To achieve this, you can utilize the [emojiPickerSettings](../api/rich-text-editor/emojiPickerSettings/) property.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:20]: By default, a predefined set of emojis is configured. However, you can customize these icons according to your needs. To achieve this, utilize the [emojiPickerSettings](../api/rich-text-editor/emojiPickerSettings/) property.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+{% include code-snippet/rich-text-editor/emoji-picker-cs1/index.ts %}
+{% endhighlight %}
+{% highlight html tabtitle="index.html" %}
+{% include code-snippet/rich-text-editor/emoji-picker-cs1/index.html %}
+{% endhighlight %}
+{% endtabs %}
+          
+{% previewsample "page.domainurl/code-snippet/rich-text-editor/emoji-picker-cs1" %}
+
+> RichTextEditor features are segregated into individual feature-wise modules. To use emoji picker, inject the Emoji Picker module using the `RichTextEditor.Inject(EmojiPicker)`.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:115]: > RichTextEditor features are segregated into individual feature-wise modules. To use emoji picker, inject the Emoji Picker module using the `RichTextEditor.Inject(EmojiPicker).`

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 23]
```diff
@@ -0,0 +1,152 @@
...
+
+## Using the shortcut key to open the emoji picker
+
+You can quickly access the emoji picker by simply pressing the colon (:) key while typing a word prefix in an editor, allowing for instant emoji selection and display. Moreover, you can continue typing in the editor after the colon (:) to filter and refine your search for the desired emojis.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:119]: Quickly access the emoji picker by pressing the colon (:) key while typing a word prefix in an editor, allowing instant emoji selection and display. Moreover, continue typing in the editor after the colon (:) to filter and refine your search for the desired emojis.

File: ej2-javascript/rich-text-editor/ts/emoji-picker.md [hunk 24]
```diff
@@ -0,0 +1,152 @@
...
+`Arrow keys`: Use the arrow keys (up, down, left, right) to move the emoji focus in the corresponding direction.
+
+`Enter`: Press Enter key to select the currently focused emoji.
+
+`Escape`: Press Escape to close the emoji picker popup without selecting an emoji.
+
+
+## Opening the emoji picker in a custom position
+
+You can display the emoji picker anywhere within the editor by using the `showEmojiPicker` public method. By default, it opens at the current cursor position within the editor. If you wish to customize the position of the popup by calling the `showEmojiPicker()` method and specifying the `x` and `y` parameters.
```
Comment (Chrispineagunja) [inline ej2-javascript/rich-text-editor/ts/emoji-picker.md:136]: Display the emoji picker anywhere within the editor by using the `showEmojiPicker` public method. By default, it opens at the current cursor position within the editor. If you wish to customize the position of the popup, call the `showEmojiPicker()` method and specify the `x` and `y` parameters.

**Review by Chrispineagunja — COMMENTED** [review]
Content review completed.

**Review by thangavele — APPROVED** [review]

**Review by thangavele — APPROVED** [review]

**Pull request merged**